pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -205,6 +222,27 @@ func (s *Server) watchRootCertAndGenKeyCert(stop <-chan struct{}) {
 	}
 }
 
+func (s *Server) watchDNSCertForK8sCA(stop <-chan struct{},
+	defaultCACertPath string,
+	signerName string,
+	approveCsr bool,
+	requestedLifetime time.Duration,
+) {
+	certUtil := certutil.NewCertUtil(int(defaultCertGracePeriodRatio * 100))
+	for {
+		waitTime, _ := certUtil.GetWaitTime(s.istiodCertBundleWatcher.GetKeyCertBundle().CertPem, time.Now(), time.Duration(0))
+		if !sleep.Until(stop, waitTime) {
+			return
+		}
+		certChain, keyPEM, _, err := chiron.GenKeyCertK8sCA(s.kubeClient.Kube(),
+			strings.Join(s.dnsNames, ","), defaultCACertPath, signerName, approveCsr, requestedLifetime)
+		if err != nil {
+			log.Fatalf("failed regenerating key and cert for istiod by kubernetes: %v", err)
+		}
```

Comment:
should we also have a retry, as you know it is very common to have error because of network transient  issue or ratelimited

Reply:
Retries are already being handled in GenKeyCertK8sCA while submitting CSR and getting signed certificate.

Reply:
I am concerned the fatal will cause process exit

Reply:
got it...I can change it to log.Errorf to not exit the process.

Reply:
I changed the log level, @hzxuzhonghu Please have a look.

Reply:
Still need continue

Reply:
oops..sorry, missed it. @hzxuzhonghu  Added it now, thank you.